lighthouse-core/gather/gather-runner.js

```diff
@@ -427,6 +429,28 @@ class GatherRunner {
     };
   }
 
+  /**
+   * Uses the debugger protocol to fetch the manifest from within the context of
+   * the target page, reusing any credentials, emulation, etc, already established
+   * there.
+   *
+   * Returns the parsed manifest or null if the page had no manifest. If the manifest
+   * was unparseable as JSON, manifest.value will be undefined and manifest.warning
+   * will have the reason. See manifest-parser.js for more information.
+   *
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts['Manifest']>}
+   */
+  static async getWebAppManifest(passContext) {
+    // If we already have a manifest, return it.
+    if (passContext.baseArtifacts.WebAppManifest) return passContext.baseArtifacts.WebAppManifest;
```

Comment:
not needed with the `isFirstPass` check below?

Reply:
fair, removed